### DISCUSSION TITLE
WebSocket Issues After 2.3.61 Upgrade

### DISCUSSION THREAD

**Discussion by wdhachfeld** [Security-Onion-Solutions/securityonion#4914] [Unsupported Versions] [answered]
I've been using a standalone 2.3.52 installation to monitor my small home network for about 2 months now. Primarily to keep an eye on the various IoT devices on my LAN for any sign of trouble. Absolutely loving the ability to dig into any questionable activity. Yesterday I performed the upgrade to 2.3.61 and it was relatively uneventful. I've successfully setup Elastic authentication and downsampled my old Grafana data.

Since upgrading, however, when I connect to the SOC via Safari on my Mac workstation I see two messages that I didn't see previously. I'm getting the following in the top UI bar:

> Attempting to connect to manager

And just below that a red box with the message:

> The operation is insecure.

Everything _seems_ to work fine. But obviously the messages are concerning. They did not appear when connecting via Safari previous to the 2.3.61 upgrade. And if I connect using Google Chrome from the same system I see neither message.

My search of these discussions yielded no hits for the later message. And only one significant hint for the former. That discussion suggested that it indicates a WebSocket. Delving into the page using Safari's Web Inspector I see the following message in the alerts:

> Refused to connect to wss://securityonion.lan/ws because it appears in neither the connect-src directive nor the default-src directive of the Content Security Policy.

A couple more web searches indicates this is probably related to the [WebKit Refined Content Security Policy](https://webkit.org/blog/6830/a-refined-content-security-policy/). And checking the header for the SOC page I actually don't see any meta element containing:

`<meta http-equiv="Content-Security-Policy" ...`

So I'm imagining this is likely the issue. Was there a change to the SOC that might have caused this? Anything I can do to eliminate the message (sans changing browser)?

Thanks in advance for any assistance you can provide!

**Comment by TOoSmOotH** [comment, marked as answer]
Yes, there was an issue with cyberchef that required us to change the CSP in .61.  Technically we only support Chromium-based browsers which will give you the optimal experience.

**Reply by wdhachfeld** [reply]
Understood. It would be great if this could be address if/when the related cyber chef issue is addressed. But obviously you guys can only support so many OS, browser, etc. combinations. I'll switch over to using Chrome for now when accessing my SOC. Thanks for the reply, Mike!

**Reply by dougburks** [reply]
Security Onion 2.3.61 Hotfix for STENO and CSP Now Available!
https://blog.securityonion.net/2021/07/security-onion-2361-hotfix-for-steno.html

**Reply by wdhachfeld** [reply]
Many thanks, @TOoSmOotH, @dougburks, and @jertel for the quick fix for this!

I updated my SO via soup, restarted the SOC container, flushed my browser caches, and I am again able to connect to the SOC via Safari without either warning message appearing. It works again just as it had with 2.3.52.

You guys rock! Thanks again!
